sdk/main/src/AccountId.cc

```diff
@@ -41,9 +41,10 @@ AccountId::AccountId(const uint64_t& shard, const uint64_t& realm, const uint64_
 //-----
 AccountId::AccountId(const std::string& str)
 {
+  size_t numDots = 0ULL;
   for (char c : str)
   {
-    if (!isdigit(c) && c != '.')
+    if ((!isdigit(c) && c != '.') || (c == '.' && ++numDots > 2))
```

Comment:
I would suggest removing this `numDots` check in the loop, and just asserting below the loop that `numDots == 2`

As it currently stands, the following test assertion *should* throw an error, but doesn't:
```
EXPECT_THROW(AccountId(testShardNumStr + testRealmNumStr + '.' + testAccountNumStr), std::invalid_argument);
```

Reply:
It would throw an error because it checks to make sure that the `shardStr`, `realmStr`, and `accountStr` aren't empty, which `realmStr` would be since the substring between `firstDot` and `secondDot` (since they're the same dot) would be size=0. This is a lot shoved into one line of code though, I can spread it out a bit.

Reply:
I tried the test and it didn't fail - nothing is thrown by that input

Reply:
Sorry yeah I was saying the constructor would throw, therefore not causing the test to fail.

Reply:
But it doesn't throw? Unless I'm misunderstanding what you're saying

Calling `AccountId("66.77")` succeeds, with a resulting shard, realm, account number of `66, 77, 77`

Reply:
Oh no I totally misunderstood what you were saying, my bad. Yeah you are correct, that case is not caught. I'll update the test with that and directly check for two dots.